drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/ScenarioSimulationServiceImpl.java

```diff
@@ -394,13 +396,17 @@ protected void ensureDependencies(KieModule module) {
             toSave |= removeFromPomIfNecessary(dependencies, oldDependency);
         }
 
+        boolean activatorToRecreate = toSave;
```

Comment:
Related to single responsibility of the method. I think in this method we should name the variable independently from `activator`. I think it is just side effect we remove `activator` according the change of dependencies. What do you think?

Reply:
Correct, I renamed the variable to `oldDependenciesExist` and added a javadoc